sensory_jamming.py

```diff
@@ -0,0 +1,342 @@
+            # time of the simulation, in seconds. Initially set at 0.
+        self.timeclock = np.empty(self.simduration, dtype = float)
+            # empty array for floats, dimensions 1*simduration. Will 
+            # contain all times in seconds for each simulation time step.
+        self.timeclock[0] = float(self.realtime) 
+            # record the first time, i.e. 0.
+        self.realduration = float(self.simduration * self.d_t)
+            # duration of the simulation in seconds.
+        self.all_ID = np.empty(self.popsize, dtype = int)
+            # empty array for integers, dimensions 1*popsize. Will 
+            # contain the identification numbers of all the agents to be run.
+        self.all_initpos = np.empty([self.popsize,2], dtype = float)
+            # empty array of intial positions for the whole population.
+        self.callsources = {}
+            # empty dictionary. Will contain the sources of each calls, 
+            # emitted by every agent throughout the simulation.
+        self.boxsize = boxsize 
+            # list of 2 integers. Area (rectangle) within which the agents can 
+            # move, defined by the lengths of edges in meters. 
+            # Has to be of the form [x,y], which is asserted below:
+        
+        assert isinstance(self.boxsize, list) and len(self.boxsize) == 2, "'boxsize' must be a list of 2 elements." 
+            # returns error if boxsize is not of the right format, i.e. [x,y]
+        
+    def Identification(self):
+        
+        for agent in range(self.popsize):
+            # for each agent in the bat population:
+            self.all_ID[int(agent)] = agent
+                # fill-in a list of all agents ID 
+                # from 0 to n = (population size - 1)
+            
+    def Positions(self):
+        
+        for ID in self.all_ID:
+            # for each agent in the bat population:
+           self.agent_initx = rd.uniform(0, self.boxsize[0])
+               # give each agent a random coordinate on the x-axis, 
+               # within the environment boundaries
+           self.agent_inity = rd.uniform(0, self.boxsize[1])
+               # give each agent a random coordinate on the y-axis, 
+               # within the environment boundaries
+           self.all_initpos[int(ID),0] = self.agent_initx
+               # store the x-coordinate in an array
+           self.all_initpos[int(ID),1] = self.agent_inity
+               # store the y-coordinate in an array
+    
+    def Timeline(self):
+            
+        for iteration in range(self.simduration):
+            # for each time step in the simulation:
+            self.realtime += float(self.d_t)
+                # increment timecount with time resolution. 
+            self.timeclock[int(iteration)] = self.realtime
+                # store the new, incremented time count in timeclock.
+
+    class Bat_Jamming_00:
+        # Class implementation nested into Launcher class implementation.
```

Comment:
this is pretty cool. Does this mean you can have multiple instances of 'Launcher' running on your computer (with each instance having their own instance of the 'Bat_Jamming_00') ?